### PR TITLE
Create hash table based on primary or secondary proc

### PR DESCRIPTION
**Pull request by dennisafa**
As pointed out by @koolzz, we should only call rte_hash_create() from the primary process. NF's would call onvm_ft_create() and call rte_hash_create() as a secondary process, which DPDK does [not allow](https://doc.dpdk.org/api/rte__hash_8h.html#a5afbd2564f738149a241bc22b2428612)

## Summary:
When onvm_ft_create is called, whether from an NF or the manager, we now check what process type it is being called from with [rte_eal_process_type()](https://doc.dpdk.org/api/rte__eal_8h.html#a1280c4f5e0f2082163ecb1f30a968cd9) If it's a secondary process (called from an NF) then we'll make the manager initialize our hash_table using a mechanism that is similar to @AaronCoplan 's firewall LPM request method. 

**Usage:**
Run the manager, then run any NF that calls onvm_ft_create. flow_tracker is one example. 

| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  
| Breaking API changes     |
| Internal API changes     | X
| Usability improvements   |  
| Bug fixes                | X
| New functionality        |
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 

## Merging notes:
 - Dependencies: None  

**TODO before merging :**
 - [x] PR is ready for review

## Test Plan:
Run the manager, then run the flow_tracker or the load_balancer NF and test its functionality.

## Review: 
@koolzz @kevindweb

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

File: onvm/onvm_nflib/onvm_common.h [hunk 1]
```diff
@@ -361,6 +362,11 @@ struct lpm_request {
         int status;
 };
 
+struct ft_request {
```
Comment (koolzz) [inline onvm/onvm_nflib/onvm_common.h]: Drop a comment for his it's used

File: onvm/onvm_nflib/onvm_nflib.h [hunk 2]
```diff
@@ -270,6 +270,9 @@ onvm_nflib_scale(struct onvm_nf_scale_info *scale_info);
 int
 onvm_nflib_request_lpm(struct lpm_request *req);
 
+int
```
Comment (koolzz) [inline onvm/onvm_nflib/onvm_nflib.h]: Add a comment

**Comment by twood02** [comment]
Does the current code call the hash create functions in a secondary process?  Does this mean that our current flow table code just doesn't work?

**Comment by koolzz** [comment]
@twood02 yes currently it's called from secondary process, it works but dpdk says that it's not okay

**Comment by dennisafa** [comment]
yes. also when created from a secondary process hash_create sporadically fails and throw a `rte_hash_create failed: could not create hash table`

**Comment by dennisafa** [comment]
> Lgtm, do we have a NF that uses the flow table so we can test this?

yes, flow_tracker [uses it](https://github.com/sdnfv/openNetVM/blob/master/examples/flow_tracker/flow_tracker.c#L236). pktge

**Comment by kevindweb** [comment]
So just so I'm on the same page, instead of calling rte_hash_create from the [subprocess here](https://github.com/sdnfv/openNetVM/pull/159/files#diff-eafd89ef7482cf7daf926de3fa882038L74) you are enqueueing a message so that our [main thread](https://github.com/sdnfv/openNetVM/pull/159/files#diff-479d1b8e19af188495c8efbed03feb55R182) picks it up and creates it (because dpdk doesn't like subprocesses creating)?

**Comment by kevindweb** [comment]
Before you get back to me, I did performance test everything, Pktgen looks solid, did a few shared core speed_tester extreme tests as well and it looked clean.

**Comment by dennisafa** [comment]
> So just so I'm on the same page, instead of calling rte_hash_create from the [subprocess here](https://github.com/sdnfv/openNetVM/pull/159/files#diff-eafd89ef7482cf7daf926de3fa882038L74) you are enqueueing a message so that our [main thread](https://github.com/sdnfv/openNetVM/pull/159/files#diff-479d1b8e19af188495c8efbed03feb55R182) picks it up and creates it (because dpdk doesn't like subprocesses creating)?

yea. if rte_hash_create is called from a secondary process, manager (primary process) will create it via this message system. `RTE_PROC_TYPE` tells us what kind of process it is.

**Comment by koolzz** [comment]
@onvm CI before improve

**Comment by onvm** [comment]
> @onvm CI before improve

## CI Message

Your results will arrive shortly

**Pull request merged**